devicetypes/hongtat/cooltouch-zwave-light-switch.src/cooltouch-zwave-light-switch.groovy

```diff
@@ -0,0 +1,461 @@
+/**
+ *  Z-Wave Light Switch Multichannel - Device Handler (parent)
+ *
+ *  Copyright 2017 HongTat Tan
+ *
+ *
+ *  Requires ** Z-Wave Light Switch Multichannel Child Device **
+ *
+ *  Tested on the following Z-Wave light switches:
+ *      NEO Coolcam Z-Wave Light Switch (EU-1/2 Gang)
+ *      MCOHome Z-Wave Light Switch (MH-312-EU)
+ *
+ *
+ *  Version history:
+ *      1.0.3 (7/10/2017) - Add 2-way status reporting
+ *      1.0.2 (25/09/2017) - Add Health Check & Configure for child device
+ *      1.0.1 (23/09/2017) - Bug fix
+ *      1.0 (23/09/2017) - Initial Release
+ *
+ *  Incorporates code from:
+ *      Peter Major  - https://github.com/petermajor/
+ *      Omar Shahbal - https://github.com/omarshahbal/
+ *      Eric Maycock - https://github.com/erocm123/
+ *      David Lomas - https://github.com/codersaur/SmartThings/tree/master/devices/zwave-tweaker
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+metadata {
+    definition (name: "COOLTOUCH Z-Wave Light Switch", namespace: "hongtat", author: "HongTat Tan") {
+    capability "Actuator"
+    capability "Switch"
+    capability "Refresh"
+    capability "Health Check"
+    capability "Polling"
+    capability "Configuration"
+    capability "Sensor"
+    capability "Zw Multichannel"
+
+    attribute "lastCheckin", "String"
+
+    preferences {
+        section(title: "Check-in Interval") {
+            paragraph "Run a Check-in procedure every so often."
+            input "checkin", "enum", title: "Run Check-in procedure", options: ["Every 1 minute", "Every 5 minutes", "Every 10 minutes", "Every 15 minutes", "Every 30 minutes", "Every 1 hour"], description: "Allows check-in procedure to run every so often", defaultValue: "Every 1 minute", required: true
+        }
+        section(title: "Check-in Info") {
+            paragraph "Display check-in info"
+            input "checkinInfo", "enum", title: "Show last Check-in info", options: ["Hide", "MM/dd/yyyy h:mm", "MM-dd-yyyy h:mm", "dd/MM/yyyy h:mm", "dd-MM-yyyy h:mm"], description: "Show last check-in info.", defaultValue: "dd/MM/yyyy h:mm", required: true
+        }
+    }
+
+    fingerprint cc: "20,25,85,8E,59,71,55,86,98,60,5A", deviceJoinName: "COOLTOUCH Z-Wave Light Switch"
```

Comment:
You need to provide device specific fingerprint with `mfr`, `prod` and `model` specified, instead of this generic.

Your fingerprint will look probably like this:
`fingerprint mfr:"<mfr of your device>", prod:"<prod of your device>", model:"<model of your device>", deviceJoinName: "COOLTOUCH Switch 1", mnmn: "SmartThings", vid: "generic-switch" //COOLTOUCH Z-Wave Light Switch`

Keep in mind that I still mean that it has to be provided while editing `zwave-multi-metering-switch`.